### ISSUE TITLE
Use DateTimeOffset in C# in favor of DateTime

### ISSUE THREAD

**Issue by schani** [quicktype/quicktype#502]
`DateTime` does not include time zone information.

**Comment by schani** [comment]
Unfortunately when parsing a naked date (no time or time-zone specified), it seems like it's parsed into a local time `DateTimeOffset`, and then no matter what I specify as `DateTimeZoneHandling`, the time-zone always shows up as local, which does not compare to the original date.

For example, in `us-senators.json` we have Bernie Sanders' birthday:

```json
    "birthday": "1941-09-08",
```

which gets serialized, and when deserializing it again we get

```json
    "birthday": "1941-09-08T00:00:00-08:00",
```

See [this commit](https://github.com/quicktype/quicktype/commit/016675615690a47beb391fe6343c2fa0c7c62259)

**Comment by krk** [comment]
When a `DateTime` or a `DateTimeOffset` is parsed from a string that doesn't specify time zone information, its `Kind` is left as `DateTimeKind.Unspecified`. In order to parse it as a UTC date, consider specifying a style:

```csharp
DateTimeOffset.TryParse("1941-09-08", DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal, out var dt);
```

**Comment by schani** [comment]
@krk implemented this in #540
